syntax-markdown-flavor/syntax-markdown-flavor-github/src/main/java/org/xwiki/contrib/rendering/markdown/flavor/github/internal/DefaultMarkdownGithubConfiguration.java

```diff
@@ -0,0 +1,110 @@
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.rendering.markdown.flavor.github.internal;
+
+import com.vladsch.flexmark.ext.abbreviation.AbbreviationExtension;
+import com.vladsch.flexmark.ext.autolink.AutolinkExtension;
+import com.vladsch.flexmark.ext.definition.DefinitionExtension;
+import com.vladsch.flexmark.ext.gfm.strikethrough.StrikethroughSubscriptExtension;
+import com.vladsch.flexmark.ext.superscript.SuperscriptExtension;
+import com.vladsch.flexmark.ext.tables.TablesExtension;
+import com.vladsch.flexmark.ext.wikilink.WikiLinkExtension;
+import com.vladsch.flexmark.ext.xwiki.macros.MacroExtension;
+import com.vladsch.flexmark.parser.Parser;
+import com.vladsch.flexmark.parser.ParserEmulationProfile;
+import com.vladsch.flexmark.util.data.MutableDataHolder;
+import com.vladsch.flexmark.util.data.MutableDataSet;
+import com.vladsch.flexmark.util.misc.Extension;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.contrib.rendering.markdown.markdown12.MarkdownConfiguration;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * GitHub Flavored Markdown Configuration.
+ *
+ * @version $Id$
+ * @since 8.7
+ */
+@Component
+@Singleton
+@Named("MarkdownConfigurationGithub")
+public class DefaultMarkdownGithubConfiguration implements MarkdownConfiguration
+{
+    private static final List<Class<?>> DEFAULT_EXTENSIONS = Arrays.asList(
+            WikiLinkExtension.class,
+            AutolinkExtension.class,
+            DefinitionExtension.class,
+            TablesExtension.class,
+            StrikethroughSubscriptExtension.class,
+            SuperscriptExtension.class,
+            AbbreviationExtension.class,
+            MacroExtension.class
+    );
+
+    @Inject
+    private Logger logger;
+
+    @Override
+    public MutableDataHolder getOptions()
+    {
+        return getDefaultOptions();
+    }
+
+    /**
+     * Creates and returns options with GitHub ParserEmulationProfile.
+     *
+     * @return the configured options for GitHub Flavor.
+     */
+    public MutableDataHolder getDefaultOptions()
+    {
+        // Configure Parser Family
+        MutableDataHolder options = new MutableDataSet();
+        options.setFrom(ParserEmulationProfile.GITHUB);
```

Comment:
If I'm correct everything is similar as in `DefaultMarkdownConfiguration` except this part. So in order to avoid the code duplication, you should only have to specify this ParserEmulationProfile. Maybe what you can do is to define in `DefaultMarkdownConfiguration` a method to get the ParserEmulationProfile and make this class inherits from it so it could override this method.

Reply:
If I use that instead, I would no more need `@Named` injection anymore right?

Reply:
>  I would no more need @Named injection anymore right?

You actually would still need it.
What change in my proposal is just that you avoid to duplicate code between `DefaultMarkdownConfiguration` and `DefaultMarkdownGithubConfiguration`.

Reply:
But we are now extending `DefaultMarkdownConfiguration` which would inherit everything except the one we would override (which is the `getOptions()` method) to minimize code duplication and only specify `ParserEmulationProfile`. So I don't think we need `@Named` notation anymore but using `MarkdownGithubConfiguration` simply (since `@Named` would be used if 2 classes are implementing the same interface, i.e. `MarkdownConfiguration` , which is no more the case).
I am pushing a commit to make it clear what I mean. I might be wrong, but let's see.

Reply:
> (since @Named would be used if 2 classes are implementing the same interface, i.e. MarkdownConfiguration , which is no more the case)

That's actually exactly what will happen. You will get `DefaultMarkdownConfiguration` directly implementing `MarkdownConfiguration` and `DefaultGithubMarkdownConfiguration` also implementing it, because it inherits from `DefaultMarkdownConfiguration`. It's actually an easy way to create a new implem of component and you already used it in your PR for example there: https://github.com/xwiki-contrib/syntax-markdown/pull/1/files#diff-66886489a9b29149e00a540b776dba7bR40

Reply:
Ah, now I see. I tested just extending `DefaultMarkdownConfiguration` and it threw an Exception that couldn't render the page. So then, I extended it + implemented `MarkdownConfiguration` and used `@Named` for the new `MarkdownGithubConfiguration` implementation of it. Then it worked fine. 
Sending a commit now. I still may be wrong, so thanks a lot for all the help you are providing.